src/components/RecordFormStateToggle/index.tsx

```diff
@@ -30,36 +30,33 @@ function RecordFormStateToggle({
   onClick,
   requireConfirm,
   message = 'Are you sure?',
-  value: inputValue,
+  value = FORM_VARIANT.VIEW,
   allowEdit,
 }: RecordFormStateToggleProps) {
-  const [value, setValue] = useState(inputValue);
   const [dialogOpen, setDialogOpen] = useState(false);
-
-  useEffect(() => {
-    setValue(inputValue);
-  }, [inputValue]);
+  const nextValue = useRef(value);
```

Comment:
I used ref because the value doesn't matter for rendering. it's only to store what was clicked somewhere while the user goes through the confirm dialog